utilities/fault_injection_fs.cc

```diff
@@ -99,7 +99,8 @@ IOStatus TestFSWritableFile::Append(const Slice& data, const IOOptions&,
   state_.buffer_.append(data.data(), data.size());
   state_.pos_ += data.size();
   fs_->WritableFileAppended(state_);
-  return IOStatus::OK();
+  IOStatus io_s = fs_->InjectWriteError(state_.filename_);
```

Comment:
is there an assumption that our retry logic must always throw away the file? (If it doesn't, I believe retrying on a same file would result in the data being double appended.)

Reply:
Yes, the flush and compaction will remove the obsolete files.